rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportService.java

```diff
@@ -109,4 +143,168 @@ public ByteBuffer getLicenseReportStreamFromURl(String token)
     public void sendComponentExportSpreadsheetSuccessMail(String emailURL, String email) throws TException {
         componentclient.sendExportSpreadsheetSuccessMail(emailURL, email);
     }
+
+    public ByteBuffer getLicenseResourceBundleBuffer() throws TException {
+        return licenseClient.getLicenseReportDataStream();
+    }
+
+    public ByteBuffer downloadSourceCodeBundle(String projectId, HttpServletRequest request, User sw360User)
+            throws TException {
+        Map<String, Set<String>> selectedReleaseAndAttachmentIds = getSelectedReleaseAndAttachmentIdsFromRequest(
+                request, false);
+        Set<String> selectedAttachmentIds = new HashSet<>();
+        selectedReleaseAndAttachmentIds.forEach((key, value) -> selectedAttachmentIds.addAll(value));
+
+        try {
+            Project project = projectclient.getProjectById(projectId, sw360User);
+            saveSourcePackageAttachmentUsages(project, sw360User, selectedReleaseAndAttachmentIds);
+            List<AttachmentContent> attachments = new ArrayList<>();
+            for (String id : selectedAttachmentIds) {
+                attachments.add(attachmentClient.getAttachmentContent(id));
+            }
+            return serveAttachmentBundle(attachments, request, project, sw360User);
+        } catch (TException e) {
+            throw new TException(e.getMessage());
+        }
+    }
+
+    public static Map<String, Set<String>> getSelectedReleaseAndAttachmentIdsFromRequest(HttpServletRequest request,
+            boolean withPath) {
+        Map<String, Set<String>> releaseIdToAttachmentIds = new HashMap<>();
+        String[] checkboxes = request.getParameterValues("licenseInfoAttachmentSelected");
+        if (checkboxes == null) {
+            return ImmutableMap.of();
+        }
+        Arrays.stream(checkboxes).forEach(s -> {
+            String[] split = s.split(":");
+            if (split.length >= 2) {
+                String attachmentId = split[split.length - 1];
+                String releaseIdMaybeWithPath;
+                if (withPath) {
+                    releaseIdMaybeWithPath = Arrays.stream(Arrays.copyOf(split, split.length - 1))
+                            .collect(Collectors.joining(":"));
+                } else {
+                    releaseIdMaybeWithPath = split[split.length - 2];
+                }
+                if (!releaseIdToAttachmentIds.containsKey(releaseIdMaybeWithPath)) {
+                    releaseIdToAttachmentIds.put(releaseIdMaybeWithPath, new HashSet<>());
+                }
+                releaseIdToAttachmentIds.get(releaseIdMaybeWithPath).add(attachmentId);
```

Comment:
Can be simplified as
```suggestion
                releaseIdToAttachmentIds.putIfAbsent(releaseIdMaybeWithPath, new HashSet<>()).add(attachmentId);
```